ietf/community/utils.py

```diff
@@ -29,15 +29,26 @@ def states_of_significant_change():
         Q(type="draft", slug__in=['rfc', 'dead'])
     )
 
-def lookup_community_list(username=None, acronym=None):
-    assert username or acronym
+class MultiplePersonError(Exception):
+    """More than one Person record matches the given email or name"""
+    pass
+
+def lookup_community_list(request, email_or_name=None, acronym=None):
+    assert email_or_name or acronym
 
     if acronym:
         group = get_object_or_404(Group, acronym=acronym)
         clist = CommunityList.objects.filter(group=group).first() or CommunityList(group=group)
     else:
-        user = get_object_or_404(User, username__iexact=username)
-        clist = CommunityList.objects.filter(person__user=user).first() or CommunityList(person=user.person)
+        persons = lookup_persons(email_or_name)
+        if len(persons) > 1:
+            if hasattr(request.user, 'person') and request.user.person in persons:
+                person = request.user.person
+            else:
+                raise MultiplePersonError("\r\n".join([p.user.username for p in persons]))
```

Comment:
I think this will just return a list of usernames, which winds up displayed as the response if a user hits this. It'd be nice to provide a bit more context in the error message.

(If you're testing this in the dev browser, that often provides a much richer response when it hits an exception than happens in production)

Reply:
Yeah, I'm not really happy with it, but it was copied from Henrik's code in `person.views.photo`. I played with the idea of returning a selection list, or of showing all matching community lists on one page (as in `person.views.profile`), but ultimately punted in favor of preferring the logged-in user - if I'm John Doe, I probably want my community list rather than any other John Doe's. Ultimately, this is (hopefully) an extreme edge case.

Reply:
That makes sense - I'm just imagining something like "Found multiple matches: " at the start of the error message to give the response some meaning for a poor user who happens upon this.